packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -749,3 +754,38 @@ export function reactHasJsxRuntime(): boolean {
 
   return false
 }
+
+export function ensureRequireEslintRules(config: Configuration): Configuration {
+  // for fast refresh we want to ensure that that there is eslint rule running
+  // because user might have added their own `eslint-loader` let's check if there is one
+  // and adjust it to add the rule or append new loader with required rule
+  const rule = config.module.rules.find(rule => {
+    if (typeof rule.loader === `string`) {
+      return rule.loader.includes(`eslint-loader`)
```

Comment:
Includes? No risk of false positives here?

Reply:
maybe I can keep `includes('eslint-loader')` as "precheck" and if those are satisfied do a deeper check (i.e. read package.json of imported package and see if package actually is `eslint-loader`), but no matter how I slice it - it will be hacky :(

Reply:
I think filename based validation is fine here. But a substr check seems a little too much of a shortcut, that's all.

Reply:
Hmmm, so do you have any advice on what would good to do here?

I can cover maybe few specific scenario:
 - `loader` is defined by package name (not fully resolved) - so for this I would do strict comparison to `eslint-loader` (no includes)
 - `loader` has some resolved path (so either `eslint-loader/index.js` or `eslint-loader/dist/cjs.js`) - then I can do includes or regex matching those

Reply:
Added hopefully more strict checks in https://github.com/gatsbyjs/gatsby/pull/28911/commits/f477549ca5a241254aab003440f3358053ae8dee which I don't particularly like (but neither I did `includes('eslint-loader')`) as it's hacky no matter how I slice it